File: Visual Basic.NET/Experiment 2/Form2.vb
Imports System.Drawing.Imaging

Public Class Form2
    ' Program 2: Determine whether an inputted integer number is prime or not

    Private Sub button_exit_Click(sender As Object, e As EventArgs) Handles button_exit.Click
        Application.Exit()
    End Sub

    Private Sub button_back_Click(sender As Object, e As EventArgs) Handles button_back.Click
        Form1.Show()
        Hide()
    End Sub

    Private Sub button_prime_Click(sender As Object, e As EventArgs) Handles button_prime.Click
        Dim a As Integer = Convert.ToInt32(textbox_num.Text)
        If a = 1 Then
            MessageBox.Show(a.ToString() + " is not a prime number nor a composite number")
            Return
        End If
        Dim i As Integer = 2
        Dim prime As Boolean = True
        While i <= (a - 1)
            If (a Mod i = 0) Then
                prime = False
            End If
            i += 1
        End While
        If prime = False Then
            MessageBox.Show("Number is not prime")
        Else
            MessageBox.Show("Number is prime")
        End If
    End Sub

    Private Sub Form2_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim t As New ToolTip()
        t.SetToolTip(button_back, "Click here to go back")
        t.SetToolTip(button_prime, "Click here to check whether number is prime or not")
        t.SetToolTip(button_exit, "Click here to exit")
    End Sub
End Class

File: WindowsProgrammingVS/File Operations and IO/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace File_operations_in_dotnet
{
    public partial class Form2 : Form
    {
        public Form2()
        {
            InitializeComponent();
        }

        private void button_back_Click(object sender, EventArgs e)
        {
            Form1 f = new Form1();
            f.Show();
            Hide();
        }

        private void button_exit_Click(object sender, EventArgs e)
        {
            Environment.Exit(0);
        }

        private void button_write_Click(object sender, EventArgs e)
        {
            FileStream fs = new FileStream(textbox_filepath.Text, FileMode.Create, FileAccess.Write);
            StreamWriter sw = new StreamWriter(fs);
            sw.WriteLine(richtextbox_filewrite.Text);
            sw.Flush();
            fs.Close();
            MessageBox.Show("Content is written in file successfully");
        }

        private void Form2_Load(object sender, EventArgs e)
        {
            ToolTip t = new ToolTip();
            t.SetToolTip(button_back, "Click here to go back");
            t.SetToolTip(button_exit, "Click here exit the app..");
            t.SetToolTip(button_write, "Click here to write content to the file path");
            t.SetToolTip(button_read, "Click here to read content from the file");
            t.SetToolTip(button_find, "Click here to find the text within the file");
        }

        private void button_read_Click(object sender, EventArgs e)
        {
            FileStream fs = new FileStream(textbox_filepath.Text, FileMode.Open, FileAccess.Read);
            StreamReader sr = new StreamReader(fs);
            richtextbox_fileread.Text = sr.ReadToEnd();
            fs.Close();
        }

        private void button_find_Click(object sender, EventArgs e)
        {
            FileStream fs = new FileStream(textbox_filepath.Text, FileMode.Open, FileAccess.Read);
            StreamReader sr = new StreamReader(fs);
            String str = sr.ReadToEnd();
            int i = (str.IndexOf(richtextbox_filesearch.Text,0));
            if (i > -1) // -1 used to denote E.O.F similar to Python last index used as -1
            {
                MessageBox.Show("This word exists in file");
            }
            else
            {
                MessageBox.Show("This word does not exist in the file");
            }
        }
    }
}
